reference/conanfile/attributes.rst

```diff
@@ -1670,3 +1670,68 @@ When ``True`` it enables the new run in a subsystem bash in Windows mechanism. :
     class FooRecipe(ConanFile):
         ...
         win_bash = True
+
+.. _test_type:
+
+test_type
+---------
+
+.. warning::
+
+    Test type is an **experimental** feature subject to breaking changes in future releases.
+
+Available since: `1.44.0 <https://github.com/conan-io/conan/releases/tag/1.44.0>`_
+
+This attribute allows testing requirements and build requiments explicitly on test package.
+In Conan 2.0 the ``test_type`` attribute will be ignored, the behavior will be always explicit, so declaring ``test_type="explicit"`` will make the test recipe compatible with Conan 2.0.
+The possible values are:
+
+ - ``requires`` (default): The package being tested will be consumed as a regular requirement automatically.
```

Comment:
I still don't see the point of documenting this now. Why not documenting only the ``explicit`` approach? We don't want anyone to use ``test_type = "build_requires"`` anymore, it will still require yet another migration. 

Lets omit it.

Reply:
the possible reason to document it is that once someone encounters `test_type = "build_requires"` in the recipe, he can go to the documentation and read what does it mean. otherwise, the only choice is to read conan's source code to understand how does recipe work.
e.g. in CCI we now have:
```
grep -RnI test_type .
./recipes/libtool/all/test_package/conanfile.py:11:    test_type = "build_requires", "requires"
./recipes/autoconf-archive/all/test_package/conanfile.py:12:    test_type = "build_requires"
./recipes/openjdk/all/test_package/conanfile.py:9:    test_type = "build_requires"
./recipes/pkgconf/all/test_package/conanfile.py:10:    test_type = "build_requires", "requires"
./recipes/gettext/all/test_package/conanfile.py:10:    test_type = "build_requires", "requires"
./recipes/android-ndk/all/test_package/conanfile.py:7:    test_type = "build_requires"
./recipes/gtk-doc-stub/all/test_package/conanfile.py:9:    test_type = "requires", "build_requires"
./recipes/emsdk/all/test_package/conanfile.py:9:    test_type = "build_requires"
./recipes/autoconf/all/test_package/conanfile.py:12:    test_type = "build_requires"
./recipes/flex/all/test_package/conanfile.py:9:    test_type = "build_requires", "requires"
./recipes/automake/all/test_package/conanfile.py:13:    test_type = "build_requires", "requires"
```
from time to time, people ask what does it mean, but we cannot provide any page describing what's the meaning of this line.
if it's something that shouldn't be documented, then we will have to remove it from all the existing recipes (probably, adding a hook restricting it and allowing only `test_type = "explicit"`

Reply:
Ok then: But lets keep an eye, and remove it as soon as possible.